test/linux_target_test.go

```diff
@@ -3091,3 +3095,89 @@ int main() {
 		check(ctx, t, spec)
 	})
 }
+
+func testPrebuiltPackages(ctx context.Context, t *testing.T, testConfig testLinuxConfig) {
+	t.Run("Use pre-built packages from build context", func(t *testing.T) {
+		t.Parallel()
+		ctx := startTestSpan(ctx, t)
+
+		preBuiltSpec := &dalec.Spec{
+			Name:        "test-prebuilt-package",
+			Version:     "0.0.1",
+			Revision:    "1",
+			License:     "MIT",
+			Website:     "https://github.com/azure/dalec",
+			Vendor:      "Dalec",
+			Packager:    "Dalec",
+			Description: "Test using pre-built packages",
+			Sources: map[string]dalec.Source{
+				"hello": {
+					Inline: &dalec.SourceInline{
+						File: &dalec.SourceInlineFile{
+							Contents:    "#!/bin/sh\necho 'Hello from pre-built package'",
+							Permissions: 0o755,
+						},
+					},
+				},
+			},
+			Artifacts: dalec.Artifacts{
+				Binaries: map[string]dalec.ArtifactConfig{
+					"hello": {},
+				},
+			},
+			Tests: []*dalec.TestSpec{
+				{
+					Name: "Test that binary from pre-built package works",
+					Steps: []dalec.TestStep{
+						{
+							Command: "/usr/bin/hello",
+							Stdout: dalec.CheckOutput{
+								Contains: []string{"Hello from pre-built package"},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		testEnv.RunTest(ctx, t, func(ctx context.Context, client gwclient.Client) {
+			// This rebuilt.txt marker file will only be created if the package is rebuilt.
+			preBuiltSpec.Build.Steps = []dalec.BuildStep{
```

Comment:
I'm a little confused by the test here.
It looks like we build the package and then build the same definition again but with the original build as the passed in package state.
Wouldn't this test pass even before your changes from the PR?

I think we should build the package without the marker, then add the marker and validate that the marker does *not* exist since we aren't using the updated spec to build the package but instead are using the pre-built package.